### ISSUE TITLE
Issue with ng e2e on Gitlab CI - This version of ChromeDriver only supports Chrome version 81

### ISSUE THREAD

**Issue by 9HD** [angular/protractor#5425]
Hi there!

I am trying to set a CI/CD environment for my new angular project with Gitlab CI/CD.
So basically I create on my machine a new project :

ng new myproject --style=scss

I want to run my test ng test and ng e2e on gitlab.

this is my gitlab-ci.yml
`stages:
    - test

test:
    stage: test
    image: weboaks/node-karma-protractor-chrome:headless
    script:
        - npm install --silent
        - npm run test
        - npm run e2e
    tags:
        - docker    
`

I have done the different modification in karma conf and protractor conf in order to launch the test in ChromeHeadless.

Then I commit and push it all to gitlab but pipeline failed...

npm run test succeed as you can see it : 

`npm run test
 > electrolia@0.0.0 test /builds/William.d/electrolia
 > ng test --watch=false
 16 04 2020 14:22:46.378:INFO [karma-server]: Karma v4.4.1 server started at http://0.0.0.0:9876/
 16 04 2020 14:22:46.381:INFO [launcher]: Launching browsers Chromium_no_sandbox with concurrency unlimited
 16 04 2020 14:22:46.411:INFO [launcher]: Starting browser ChromiumHeadless
 16 04 2020 14:22:56.182:INFO [HeadlessChrome 79.0.3945 (Linux 0.0.0)]: Connected on socket Fi_g0kTaCEBWZDFsAAAA with id 2723840
 HeadlessChrome 79.0.3945 (Linux 0.0.0): Executed 0 of 3 SUCCESS (0 secs / 0 secs)
 HeadlessChrome 79.0.3945 (Linux 0.0.0): Executed 1 of 3 SUCCESS (0 secs / 0.173 secs)
 HeadlessChrome 79.0.3945 (Linux 0.0.0): Executed 2 of 3 SUCCESS (0 secs / 0.216 secs)
 HeadlessChrome 79.0.3945 (Linux 0.0.0): Executed 3 of 3 SUCCESS (0 secs / 0.242 secs)
 HeadlessChrome 79.0.3945 (Linux 0.0.0): Executed 3 of 3 SUCCESS (0.466 secs / 0.242 secs)
 TOTAL: 3 SUCCESS
 TOTAL: 3 SUCCESS`

Unfortunately npm run e2e doesn't work : 

This is the message I'm getting : 

`$ npm run e2e
 > electrolia@0.0.0 e2e /builds/William.d/electrolia
 > ng e2e
 [14:23:09] I/file_manager - creating folder /builds/William.d/electrolia/node_modules/protractor/node_modules/webdriver-manager/selenium
 [14:23:09] I/config_source - curl -o/builds/William.d/electrolia/node_modules/protractor/node_modules/webdriver-manager/selenium/chrome-response.xml https://chromedriver.storage.googleapis.com/
 [14:23:10] I/downloader - curl -o/builds/William.d/electrolia/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_81.0.4044.69.zip https://chromedriver.storage.googleapis.com/81.0.4044.69/chromedriver_linux64.zip
 [14:23:10] I/update - chromedriver: unzipping chromedriver_81.0.4044.69.zip
 [14:23:10] I/update - chromedriver: setting permissions to 0755 for /builds/William.d/electrolia/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_81.0.4044.69
 chunk {main} main.js, main.js.map (main) 60.6 kB [initial] [rendered]
 chunk {polyfills} polyfills.js, polyfills.js.map (polyfills) 141 kB [initial] [rendered]
 chunk {runtime} runtime.js, runtime.js.map (runtime) 6.15 kB [entry] [rendered]
 chunk {styles} styles.js, styles.js.map (styles) 10.1 kB [initial] [rendered]
 chunk {vendor} vendor.js, vendor.js.map (vendor) 2.99 MB [initial] [rendered]
 Date: 2020-04-16T14:23:46.648Z - Hash: 24529d52d67b0fb9d54b - Time: 15125ms
 ** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **
 : Compiled successfully.
 [14:23:49] I/launcher - Running 1 instances of WebDriver
 [14:23:49] I/direct - Using ChromeDriver directly...
 [14:23:50] E/launcher - session not created: This version of ChromeDriver only supports Chrome version 81
   (Driver info: chromedriver=81.0.4044.69 (6813546031a4bc83f717a2ef7cd4ac6ec1199132-refs/branch-heads/4044@{#776}),platform=Linux 4.19.78-coreos x86_64)
 [14:23:50] E/launcher - SessionNotCreatedError: session not created: This version of ChromeDriver only supports Chrome version 81
   (Driver info: chromedriver=81.0.4044.69 (6813546031a4bc83f717a2ef7cd4ac6ec1199132-refs/branch-heads/4044@{#776}),platform=Linux 4.19.78-coreos x86_64)`

I don't understand why, I have tried all the solution i could find.....still doesn't work

If someone can help me, it would be very helpful.

Thank you.

**Comment by 9HD** [comment]
I have a new clue : 
I compared two pipelins on my gitlab ci : 

`16 04 2020 14:48:33.952:INFO [karma-server]: Karma v4.4.1 server started at http://0.0.0.0:9876/
 16 04 2020 14:48:33.956:INFO [launcher]: Launching browsers Chromium_no_sandbox with concurrency unlimited
 16 04 2020 14:48:34.021:INFO [launcher]: Starting browser ChromiumHeadless
 16 04 2020 14:48:41.730:INFO [HeadlessChrome 79.0.3945 (Linux 0.0.0)]: Connected on socket ur4W7aKb0c0C-Ou3AAAA with id 79574412
 HeadlessChrome 79.0.3945 (Linux 0.0.0): Executed 0 of 3 SUCCESS (0 secs / 0 secs)
 HeadlessChrome 79.0.3945 (Linux 0.0.0): Executed 1 of 3 SUCCESS (0 secs / 0.182 secs)
 HeadlessChrome 79.0.3945 (Linux 0.0.0): Executed 2 of 3 SUCCESS (0 secs / 0.22 secs)
 HeadlessChrome 79.0.3945 (Linux 0.0.0): Executed 3 of 3 SUCCESS (0 secs / 0.243 secs)
 HeadlessChrome 79.0.3945 (Linux 0.0.0): Executed 3 of 3 SUCCESS (0.453 secs / 0.243 secs)`

`16 04 2020 17:04:24.937:INFO [karma-server]: Karma v4.4.1 server started at http://0.0.0.0:9876/
 16 04 2020 17:04:24.940:INFO [launcher]: Launching browsers Chromium_no_sandbox with concurrency unlimited
 16 04 2020 17:04:24.971:INFO [launcher]: Starting browser ChromiumHeadless
 16 04 2020 17:04:32.689:INFO [HeadlessChrome 80.0.3987 (Linux 0.0.0)]: Connected on socket SbiKWoCvh_RVnObbAAAA with id 75415711
 HeadlessChrome 80.0.3987 (Linux 0.0.0): Executed 0 of 3 SUCCESS (0 secs / 0 secs)
 HeadlessChrome 80.0.3987 (Linux 0.0.0): Executed 1 of 3 SUCCESS (0 secs / 0.186 secs)
 HeadlessChrome 80.0.3987 (Linux 0.0.0): Executed 2 of 3 SUCCESS (0 secs / 0.23 secs)
 HeadlessChrome 80.0.3987 (Linux 0.0.0): Executed 3 of 3 SUCCESS (0 secs / 0.256 secs)
 HeadlessChrome 80.0.3987 (Linux 0.0.0): Executed 3 of 3 SUCCESS (0.352 secs / 0.256 secs)
 TOTAL: 3 SUCCESS
 TOTAL: 3 SUCCESS`

Once it get Chrome version 79.0.3945 and Once 80.0.3987 but never 81.0.4044.69. 
I ran it on shared runner provided by gitlab.

Do you think the problem comes from it?

**Comment by johannesjo** [comment]
Where you able to solve your issue?

**Comment by 9HD** [comment]
Yes in fact I noticed that according the docker image I use it gets a version stable, latest or another one  where the version number could be different.
But ng e2e will execute webdriver-manager update as default and will get the latest version which could be different, I don't remember exactly sorry but the problem comes from that.
So what I did, I focused on the version of chromium use by my previous docker image and in my package.json i forced the driver like this :

`
```json
    "pree2e": "webdriver-manager update --versions.chrome 80.0.3987.106",
    "e2e": "ng e2e --webdriver-update=false"
````

my .yml file for the test stage : 
`
```yaml
test:
    stage: test
    image: weboaks/node-karma-protractor-chrome:headless
    script:
        - npm install --silent
        - npm run test
        - npm run pree2e
        - npm run e2e
    tags:
        - docker 
````

I put this problem aside for now but this is the workaround i found.
